lazy_theta_star_planner/src/lazy_theta_star2.cpp

```diff
@@ -0,0 +1,257 @@
+      curr_data.closed = -1;
+      id curr_par = curr_data.parent_id;
+
+      if (isGoal(curr_data.x, curr_data.y)) {
+        if (!(losCheck(curr_data.x, curr_data.y, data[curr_par].x, data[curr_par].y))) {
+          resetParent(curr_data);
+        }
+        break;
+      }
+
+      if (!(losCheck(curr_data.x, curr_data.y, data[curr_par].x, data[curr_par].y))) {
+        resetParent(curr_data);
+      }
+
+      setNeighbors(curr_data);
+
+      curr_id = pq.top().pos_id;
+      pq.pop();
+    }
+    if (pq.empty()) {
+      RCLCPP_INFO(node_->get_logger(), "No Path Found !!!!!!!!!!!");
+      raw_path.clear();
+      return false;
+    }
+
+  } else {
+    RCLCPP_INFO(node_->get_logger(), "Straight Line Path Found!");
+    data.push_back({dst.x, dst.y, dist(src.x, src.y, dst.x, dst.y), 0, 0, 1,
+        dist(src.x, src.y, dst.x, dst.y)});
+    curr_id = id_gen;
+  }
+
+  RCLCPP_INFO(node_->get_logger(), "REACHED DEST  %i,   %i", data[curr_id].x, data[curr_id].y);
+
+  backtrace(raw_path, curr_id);
+
+  return true;
+}
+
+void LazyThetaStar::initializePosn()
+{
+  posn.clear();
+  for (int i = 0; i < sizeX * sizeY; i++) {
+    posn.push_back(-1);
+  }
+}
+
+bool LazyThetaStar::losCheck(map_pts & x0, map_pts & y0, map_pts & x1, map_pts & y1) const
+{
+  int dy = abs(y1 - y0), dx = abs(x1 - x0), f = 0;
+  int sx = x1 > x0 ? 1 : -1, sy = y1 > y0 ? 1 : -1;
+  int cx = x0, cy = y0;
+  int ux = (sx - 1) / 2, uy = (sy - 1) / 2;
+
+  if (dx >= dy) {
+    while (cx != x1) {
+      f += dy;
+      if (f >= dx) {
+        if (!isSafe(cx + ux, cy + uy)) {
+          return false;
+        }
+        cy += sy;
+        f -= dx;
+      }
+      if (f != 0 && !isSafe(cx + ux, cy + uy)) {
+        return false;
+      }
+      if (dy == 0 && !isSafe(cx + ux, cy) && !isSafe(cx + ux, cy - 1)) {
+        return false;
+      }
+      cx += sx;
+    }
+  } else {
+    while (cy != y1) {
+      f = f + dx;
+      if (f >= dy) {
+        if (!isSafe(cx + ux, cy + uy)) {
+          return false;
+        }
+        cx += sx;
+        f -= dy;
+      }
+      if (f != 0 && !isSafe(cx + ux, cy + uy)) {
+        return false;
+      }
+      if (dx == 0 && !isSafe(cx, cy + uy) && !isSafe(cx - 1, cy + uy)) {
+        return false;
+      }
+      cy += sy;
+    }
+  }
+  return true;
+}
+
+void LazyThetaStar::backtrace(std::vector<coords<world_pts>> & raw_points, id & curr_id)
+{
+  std::vector<coords<world_pts>> path_rev;
+  do {
+    coords<world_pts> world;
+    costmap_->mapToWorld(data[curr_id].x, data[curr_id].y, world.x, world.y);
+    path_rev.push_back({world.x, world.y});
+    curr_id = data[curr_id].parent_id;
+  } while (curr_id != 0);
+  coords<world_pts> w;
+  costmap_->mapToWorld(data[curr_id].x, data[curr_id].y, w.x, w.y);
+  path_rev.push_back({w.x, w.y});
+
+  for (int i = path_rev.size() - 1; i >= 0; i--) {
+    raw_points.push_back(path_rev[i]);
+  }
+}
+
+void LazyThetaStar::resetParent(tree_node & curr_data)
```

Comment:
What's this function do?

Reply:
Lazy checking if the parent allocated earlier is in sight or not

Reply:
So this is core to the lazy theta * algorithm? OK. I think some comments are in need to describe whats happening throughout the code.